runtime/src/snapshot_utils.rs

```diff
@@ -996,6 +1540,75 @@ pub fn bank_to_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
     Ok(package.tar_output_file)
 }
 
+/// Convenience function to create an incremental snapshot archive out of any Bank, regardless of
+/// state.  The Bank will be frozen during the process.
+///
+/// Requires:
+///     - `bank` is complete
+///     - `bank`'s slot is greater than `full_snapshot_slot`
+pub fn bank_to_incremental_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
+    snapshot_path: P,
+    bank: &Bank,
+    full_snapshot_slot: Slot,
+    snapshot_version: Option<SnapshotVersion>,
+    snapshot_package_output_path: Q,
+    archive_format: ArchiveFormat,
+    thread_pool: Option<&ThreadPool>,
+    maximum_snapshots_to_retain: usize,
+) -> Result<PathBuf>
+where
+    P: fmt::Debug,
+{
+    assert!(bank.is_complete());
+    assert!(bank.slot() > full_snapshot_slot);
+
+    bank.squash(); // Bank may not be a root
```

Comment:
Bank should already be rooted since AccountsBackgroundService only gets requests to snapshot roots?

Reply:
I was wondering the same thing myself. The first and last lines are copied from the non-incremental version, and I just left them. Maybe these lines were written before ABS existed? Should either/both of these line be removed/replaced with something else instead?

Reply:
This function (well, it's FSS version, `bank_to_snapshot_archive()`) is called in two places: (1) In `Validator::new()` before `AccountsBackgroundService` is created, and (2) in `ledger-tool` with the `create-snapshot` option. So neither have ABS.

In the `Validator::new()` path, the bank _does_ get rooted before calling `bank_to_snapshot_archive()`. In the `ledger-tool` path, the bank does _not_ get rooted before calling `bank_to_snapshot_archive()`. So I think the `bank.squash()` line exists because of `ledger-tool`. Do you think this is OK?